tools/amd_build/pyHIPIFY/hipify-python.py

```diff
@@ -0,0 +1,967 @@
+                                        new_kernel_launch = re.sub(r'\b(%s)\b' %
+                                                                   arg, lambda x: static_argument, new_kernel_launch)
+
+                            # Add template type
+                            if "THCUNN" in filepath.split("/") and "generic" not in filepath.split("/"):
+                                kernel_name_with_template = kernel_name_with_template.replace("<real>", "<Dtype>")
+                            new_kernel_launch = re.sub(r'\b(%s)\b' % kernel_name,
+                                                       lambda x: kernel_name_with_template, new_kernel_launch)
+
+                            # Replace Launch
+                            new_output_source = new_output_source.replace(old_kernel_launch, new_kernel_launch)
+
+                    # Overwrite file contents
+                    fileobj.seek(0)
+                    fileobj.write(new_output_source)
+                    fileobj.truncate()
+                    fileobj.flush()
+
+                    # Flush to disk
+                    os.fsync(fileobj)
+
+
+def main():
+    """Example invocation
+
+    python hipify.py --project-directory /home/myproject/ --extensions cu cuh h cpp --output-directory /home/gains/
+    """
+
+    parser = argparse.ArgumentParser(
+        description="The Python Hipify Script.")
+
+    parser.add_argument(
+        '--project-directory',
+        type=str,
+        default=os.getcwd(),
+        help="The root of the project.",
+        required=True)
+
+    parser.add_argument(
+        '--show-detailed',
+        type=bool,
+        default=False,
+        help="Show detailed summary of the hipification process.",
+        required=False)
+
+    parser.add_argument(
+        '--extensions',
+        nargs='+',
+        default=["cu", "cuh", "c", "cpp", "h", "in", "hpp"],
+        help="The extensions for files to run the Hipify script over.",
+        required=False)
+
+    parser.add_argument(
+        '--output-directory',
+        type=str,
+        default="",
+        help="The directory to store the hipified project.",
+        required=False)
+
+    parser.add_argument(
+        '--include-dirs',
+        nargs='+',
+        default=[],
+        help="The directories under the root that should be included.",
+        required=False)
+
+    parser.add_argument(
+        '--yaml-settings',
+        type=str,
+        default="",
+        help="The yaml file storing information for disabled functions and modules.",
+        required=False)
+
+    parser.add_argument(
+        '--add-static-casts',
+        type=bool,
+        default=False,
+        help="Whether to automatically add static_casts to kernel arguments.",
+        required=False)
+
+    args = parser.parse_args()
+
+    # Verify the project directory exists.
+    if not os.path.exists(args.project_directory):
+        print("The project folder specified does not exist.")
+        return
```

Comment:
Don't `return` here, because that will cause your Python script to exit with exit code 0, indicating success. `sys.exit(1)` is good. There are a bunch of other returns which need to be adjusted too.

Reply:
done